plugins/follow_me/follow_me_impl.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include "follow_me.h"
+#include "mavlink_include.h"
+#include "plugin_impl_base.h"
+#include "device_impl.h"
+#include "timeout_handler.h"
+#include "global_include.h"
+
+/**
+  IMPORTANT NOTE:
+    Macro FOLLOW_ME_TESTING is used to test FollowMe plugin.
+    In real scenario, GCS (DroneCore Application) doesn't provide poistion,
+    but instead, current location of the device is captured by platform-specific Location Framework.
+
+ */
+#ifndef FOLLOW_ME_TESTING
+#define FOLLOW_ME_TESTING
+#endif
+
+namespace dronecore {
+
+class FollowMeImpl : public PluginImplBase
+{
+public:
+    FollowMeImpl();
+    ~FollowMeImpl();
+
+    void init() override;
+    void deinit() override;
+
+    FollowMe::Configuration get_config() const;
+    void set_config(const FollowMe::Configuration &cfg);
+
+    FollowMe::Result start();
+    FollowMe::Result stop();
+private:
+    /* private methods */
+    /*****************************************************/
+    void process_heartbeat(const mavlink_message_t &message);
+    FollowMe::Result to_follow_me_result(MavlinkCommands::Result result) const;
+    void timeout_occurred();
+    void send_gcs_motion_report();
+
+    void receive_param_min_height(bool success, float min_height_m);
+    void receive_param_follow_target_dist(bool success, float ft_dist_m);
+    void receive_param_follow_dir(bool success, int32_t dir);
+    void receive_param_responsiveness(bool success, float rsp);
+    /*****************************************************/
+
+    /**
+     * @brief Altitude flag for Follow Target command.
+     */
+    enum class ESTCapabilities {
+        POS = 0,
+        VEL = 1,
+        ACCEL = 2,
+        ATT_RATES = 3
+    };
+
+    /**
+     * @brief Motion report info which will be emitted periodically to the Vehicle
+     */
+    struct MotionReport {
+        // GCS Position
+        int32_t lat_int;
+        int32_t lon_int;
+        float alt;
+        // Velocity in NED frame in meter / s
+        float vx;
+        float vy;
+        float vz;
+        // Acceleration in NED frame in meter/s^2 or NAN
+        float afx;
+        float afy;
+        float afz;
+        // for eph, epv
+        float pos_std_dev[3];
+
+        /**
+         * @brief MotionReport
+         * @param _lat
+         * @param _lon
+         * @param _alt
+         */
+#ifdef FOLLOW_ME_TESTING
+        // Initial GCS position is set to PX4 Home
```

Comment:
As said, testing should be done using the outside API, not from the inside.

Reply:
ok